### DISCUSSION TITLE
Discrepancies between the language specification and the playground.

### DISCUSSION THREAD

**Discussion by nicuveo** [cue-lang/cue#2368] [General] [answered]
Hi all!

I've been writing an interpreter for CUE as a hobby project, and i've stumbled onto several discrepancies between what the language specification documents and what the playground actually allows, as well as several features that are simply not documented in the reference at all.

One simple example of the latter is slices: their only mention is in the grammar description of a Primary Expression, but there is no further documentation about them. The `Slice` production rule is never given.
```
PrimaryExpr
  = Operand
  | PrimaryExpr Selector
  | PrimaryExpr Index
  | PrimaryExpr Slice
  | PrimaryExpr Arguments
  .
```

My question here is: is there a preferred way for me to report those issues in the spec? Would opening a Github issue be the right approach? Would you prefer one bulky Github issue with all of them, or one issue per topic?

**Comment by mvdan** [comment, marked as answer]
For this particular one you found, see https://github.com/cue-lang/cue/issues/772 :)

I would suggest raising any major discrepancies as bug reports. Anything minor, like a typo in the grammar or an obvious mistake, you can send a patch directly. Just make sure to search the issue tracker first, e.g. via `spec in:title`.

**Reply by nicuveo** [reply]
will do, thanks! :)

**Comment by rogpeppe** [comment]
FWIW for slice expressions in particular, I believe that it's not entirely sure whether they will remain as a feature in the core language, or be removed in favour of the [equivalent functionality](https://pkg.go.dev/cuelang.org/go@v0.5.0/pkg/list#Slice) in the list package.
